pep-0714.rst

```diff
@@ -0,0 +1,274 @@
+Python releases, Downstream Linux releases, baked into containers, virtual
+environments, etc.
+
+This puts us in an awkward position of having a bug on PyPI that cannot be fixed
+without breaking pip, due to a bug in pip, but that version of pip is old enough
+to have been widely deployed. To make matters worse, a version of pip that is
+broken in this way cannot install *anything* from PyPI once it fixes its bug,
+including installing a new, fixed version of pip.
+
+
+Rationale
+=========
+
+There are 3 main options for a path forward for fixing these bugs:
+
+1. Do not change the spec, fix the bug in pip, wait some amount of time, then
+   fix the bug in PyPI, breaking anyone using an unfixed pip such that they
+   cannot even install a new pip from PyPI.
+2. Do the same as (1), but special case PyPI so it does not emit the :pep:`658`
+   metadata for pip, even if it is available. This allows people to upgrade pip
+   if they're on a broken version, but nothing else.
+3. Change the spec to avoid the key that pip can't handle currently, allowing
+   PyPI to emit that key and a new version of pip to be released to take
+   advantage of that key.
+
+This PEP chooses (3), but goes a little further and also renames the key in the
+HTML representation.
+
+Typically we do not change specs because of bugs that only affect one particular
+implementation, unless the spec itself is at fault, which isn't the case here,
+the spec is fine and these are just genuine bugs in pip and PyPI.
+
+However, we choose to do this for 4 reasons:
+
+1. Bugs that affect pip and PyPI together represent an outsized amount of impact
+   compared to any other client or repository combination.
+2. The impact of being broken is that installs do not function, at all, rather
+   than degrading gracefully in some way.
+3. The feature that is being blocked by these bugs is of large importance to
+   the ability to quickly and efficiently resolve dependencies from PyPI with
+   pip, and having to delay it for a long period of time while we wait for the
+   broken versions of pip to fall out of use would be of detriment to the entire
+   ecosystem.
+4. The downsides of changing the spec are fairly limited, given that we do not
+   believe that support for this is widespread, so it affects only a limited
+   number of projects.
+
+
+Specification
+=============
+
+The keywords "**MUST**", "**MUST NOT**", "**REQUIRED**", "**SHALL**",
+"**SHALL NOT**", "**SHOULD**", "**SHOULD NOT**", "**RECOMMENDED**", "**MAY**",
+and "**OPTIONAL**"" in this document are to be interpreted as described in
+:rfc:`RFC 2119 <2119>`.
+
+
+Servers
+-------
+
+The :pep:`658` metadata, when used in the HTML representation of the Simple API
+**MUST** be emitted using the attribute name ``data-core-metadata``, with the
+supported values remaining the same.
+
+The :pep:`658` metadata, when used in the :pep:`691` JSON representation of the
+Simple API **MUST** be emitted using the key ``core-metdadata``, with the
```

Comment:
```suggestion
Simple API, **MUST** be emitted using the key ``core-metadata``, with the
```
Or you'll have to write another PEP renaming "metdadata" to "metadata" :D

Reply:
![](https://media.giphy.com/media/12GzK1jYCaVCV2/giphy.gif)